### ISSUE TITLE
Support macCatalyst

### ISSUE THREAD

**Issue by pokryfka** [swift-server/async-http-client#304]
AsyncHTTPClient 1.2.1 commit hash: 4b4d6605aa2e4f0c2ae3c7563795ae3bec259fff

Context:
Building AsyncHTTPClient for Mac Catalyst fails with errors:

`'sec_protocol_options_set_tls_min_version' is unavailable in Mac Catalyst`
`'sec_protocol_options_set_tls_max_version' is unavailable in Mac Catalyst`

Steps to reproduce:
1. Open `Package.swift` in Xcode
2. Select platform `My Mac (Mac Catalyst)`
3. Build

$ swift --version
Apple Swift version 5.2.4 (swiftlang-1103.0.32.9 clang-1103.0.32.53)
Target: x86_64-apple-darwin19.6.0

Operating system: macOS Catalina 10.15.6

$ uname -a
Darwin xxx 19.6.0 Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64 x86_64

Xcode 11.6 
Mac Catalyst 13.0

**Comment by Lukasa** [comment]
Yes, I think the availability guards in `TLSConfiguration` need to be updated to cover `macCatalyst`. Are you interested in doing that?

**Comment by pokryfka** [comment]
@Lukasa 

I could fix the compilation.

However I think ideally this should be accompanied with CI build tests on different platforms (iOS, macCatalyst, ...) which I dont think I could create/update on my own.

My understanding is, please correct me if I am wrong, that while the main target of AsyncHTTPClient is Linux
other platforms, including macOS and iOS its "derivatives", are also supported and as such it would be nice to test, at the very least, compilation.

**Comment by Lukasa** [comment]
There are some logistical difficulties with that in general that @tomerd will need to do some thinking about. Suffice it to say that while we agree these should be tested, for now I think it's mostly important to get the software building again.
